contrib/babelfishpg_common/src/datetime2.h

```diff
@@ -21,6 +21,9 @@
 /* Range-check a datetime */
 #define IS_VALID_DATETIME2(t)  (MIN_DATETIME2 <= (t) && (t) < END_DATETIME2)
 
+extern Datum time_datetime2(PG_FUNCTION_ARGS);
+extern Datum date_datetime2(PG_FUNCTION_ARGS);
```

Comment:
why do we need this? wouldn't it collide with PG_FUNCTION_INFO_V1(...)?

Reply:
we have done something similar previously. These are such functions:
```
extern Datum timestamptz_datetimeoffset(PG_FUNCTION_ARGS);
extern Datum timestamptz_datetime2(PG_FUNCTION_ARGS);
extern Datum timestamptz_datetime(PG_FUNCTION_ARGS);
extern Datum datetimeoffset_timestamp(PG_FUNCTION_ARGS);
```